File: test-fixtures/cross-language/ruby/realistic.rb
# SPDX-License-Identifier: MIT
class SessionRecord
  attr_accessor :mutable_field
  attr_reader :immutable_field
  attr_accessor :shared_name

  def initialize(value)
    @mutable_field = value
    @immutable_field = "fixed"
  end
end

class AuditRecord
  attr_accessor :shared_name
end

File: test-fixtures/cross-language/php/realistic.php
<?php
// SPDX-License-Identifier: MIT

class InvoiceRecord {
    public string $mutableField;
    public readonly string $immutableField;
    public static int $staticField;
    private int $privateField;
    public int $sharedName;

    public function __construct(public string $promotedField, int $privateField) {
        $this->mutableField = $promotedField;
        $this->immutableField = "fixed";
        $this->privateField = $privateField;
    }
}

class AuditRecord {
    public int $sharedName;
}
